### PR TITLE
Issue #220 fixes (#1)

### PR DESCRIPTION
**Pull request by 4GForce**
* Fixed OpenSearchOnPoeTrade: (#220)

The exact same actions/lines are done within TradeFunc_Main()

* Fixed Item.xtype not cleared

Since the xtype is not used in ItemInfo script, it wasn't properly
initialized.
I didn't add the xtype to ItemInfo since its not used in that context.

* Fixed AdvancedPriceCheck: (#220)

No need to parse the clipboard again when redirected from
AdvancedPriceCheck form submit.

### CODE REVIEW

File: resources/ahk/TradeMacro.ahk [hunk 1]
```diff
@@ -1374,7 +1372,7 @@ TradeFunc_GetMeanMedianPrice(html, payload){
 		}
 		
 		; replace "
-		StringReplace, Currency, Currency, ", , All
+		StringReplace, Currency, Currency, ", , All			; "
```
Comment (Eruyome) [inline resources/ahk/TradeMacro.ahk]: What does this change do?

File: resources/ahk/TradeMacro.ahk [hunk 2]
```diff
@@ -87,13 +87,6 @@ OpenSearchOnPoeTrade:
 	SuspendPOEItemScript = 1 ; This allows us to handle the clipboard change event
 	Send ^{sc02E}
 	Sleep 250
-	
-	TradeFunc_DoParseClipboard()
-	If (!Item.Name and TradeOpts.OpenUrlsOnEmptyItem) {
```
Comment (Eruyome) [inline resources/ahk/TradeMacro.ahk]: Why did you remove this? If you don't hover over an item this function is supposed to open the wiki frontpage. 
I also noticed it doesn't work correctly anymore because the old Item.Name is still being used, using this on Facebreakers and then without item hover opens the facebreaker wiki page both times.

File: resources/ahk/TradeMacro.ahk (hunk 1 above)
Comment (4GForce) [inline resources/ahk/TradeMacro.ahk]: Oh that part was just breaking my editor because of the " on that line
so I just added an end quote in comment   ; "

File: resources/ahk/TradeMacro.ahk (hunk 2 above)
Comment (4GForce) [inline resources/ahk/TradeMacro.ahk]: That part is not even related to the wiki, its within  OpenSearchOnPoeTrade: ...
and it was redundant since the exact same lines are are the beginning of TradeFunc_Main()
```

OpenSearchOnPoeTrade:
	Global TradeOpts, Item
	Item := {}
	SuspendPOEItemScript = 1 ; This allows us to handle the clipboard change event
	Send ^{sc02E}
	Sleep 250
	
	TradeFunc_DoParseClipboard()
	If (!Item.Name and TradeOpts.OpenUrlsOnEmptyItem) {
		TradeFunc_OpenUrlInBrowser("http://poe.trade/")
		return
	}
	
	TradeFunc_Main(true)
	SuspendPOEItemScript = 0 ; Allow Item info to handle clipboard change event
return

; Prepare Reqeust Parametes and send Post Request
; openSearchInBrowser : set to true to open the search on poe.trade instead of showing the tooltip
; isAdvancedPriceCheck : set to true If the GUI to select mods should be openend
; isAdvancedPriceCheckRedirect : set to true If the search is triggered from the GUI
; isItemAgeRequest : set to true to check own listed items age
TradeFunc_Main(openSearchInBrowser = false, isAdvancedPriceCheck = false, isAdvancedPriceCheckRedirect = false, isItemAgeRequest = false)
{	
	LeagueName := TradeGlobals.Get("LeagueName")
	Global Item, ItemData, TradeOpts, mapList, uniqueMapList, Opts
		
	TradeFunc_DoParseClipboard()
	iLvl     := Item.Level
	
	; cancel search If Item is empty
	If (!Item.name) {
		If (TradeOpts.OpenUrlsOnEmptyItem) {
			TradeFunc_OpenUrlInBrowser("https://poe.trade")
		}
		return
	}
	
```

Comment (4GForce) [inline resources/ahk/TradeMacro.ahk]: Now that you mentioned the wiki, I looked at it and I think I see the problems, nothing related to my changes.
To achieve the behavior you are looking for, the clipboard would have to be cleared before sending the ctrl+c command.
Take a quick look at the 5 comments I added in the code below

```
OpenWiki:
	IfWinActive, Path of Exile ahk_class POEWindowClass 
	{
		Global TradeOpts, Item
		SuspendPOEItemScript = 1 ; This allows us to handle the clipboard change event
		
; the ctrl+c command won't reset or change the clipboard if no item is hovered
		Send ^{sc02E}
		Sleep 250
		
; parsing same old clipboard since no item was hovered
		TradeFunc_DoParseClipboard()

; now this part can only occur at script start when the clipboard is empty ( unless an item was parsed without a name for some reason )
		If (!Item.Name and TradeOpts.OpenUrlsOnEmptyItem) {
			TradeFunc_OpenUrlInBrowser("http://pathofexile.gamepedia.com/")
			
; this return would prevent ItemInfo to resume since SuspendPOEItemScript is never setted back to 0
			return
		}
		
; I'm not sure this can cause a problem but UrlAffix is never initialized
		If (Item.IsUnique or Item.IsGem or Item.IsDivinationCard or Item.IsCurrency) {
			UrlAffix := Item.Name
		} Else If (Item.IsFlask or Item.IsMap) {
			UrlAffix := Item.SubType
		} Else If (RegExMatch(Item.Name, "i)Sacrifice At") or RegExMatch(Item.Name, "i)Fragment of") or RegExMatch(Item.Name, "i)Mortal ") or RegExMatch(Item.Name, "i)Offering to ") or RegExMatch(Item.Name, "i)'s Key") or RegExMatch(Item.Name, "i)Breachstone")) {
			UrlAffix := Item.Name
		} Else {
			UrlAffix := Item.BaseType
		}
		
		If (StrLen(UrlAffix) > 0) {			
			UrlAffix := StrReplace(UrlAffix," ","_")
			WikiUrl := "http://pathofexile.gamepedia.com/" UrlAffix		
			TradeFunc_OpenUrlInBrowser(WikiUrl)	
		}
		
		SuspendPOEItemScript = 0 ; Allow Item info to handle clipboard change event
	}
return
```

Comment (Eruyome) [inline resources/ahk/TradeMacro.ahk]: It seems I was confused about that part you deleted, mistook it for a different part...

I'm not sure this wiki problem is no related to your changes, at least to some of the more recent ones. I know it worked.

The not initialized UrlAffix shouldn't be a problem, but simply initializing it doesn't hurt.

File: resources/ahk/TradeMacro.ahk (hunk 1 above)
Comment (Eruyome) [inline resources/ahk/TradeMacro.ahk]: Thought as much, what editor are you using?

**Comment by Eruyome** [comment]
Ok, I merged it to a different branch so that i can take a better look at it and we can try to resolve the known issues before merging to master.

File: resources/ahk/TradeMacro.ahk (hunk 1 above)
Comment (4GForce) [inline resources/ahk/TradeMacro.ahk]: Notepad++

**Comment by 4GForce** [comment]
Alright, I'll triple check everything tomorrow.

File: resources/ahk/TradeMacro.ahk (hunk 1 above)
Comment (4GForce) [inline resources/ahk/TradeMacro.ahk]: I might try SciTE4AutoHotkey tomorrow.  What are you using ?

**Comment by Eruyome** [comment]
I just fixed some small things, the problem that the clipboard is not being reset still remains, this also causes ctrl +d doing a search without item hover. This has to be solved in TradeMacro (I believe), because forcing a clipboard reset in ItemInfo would prevent you from copying the item data to view/use it, for example debugging.

**Comment by 4GForce** [comment]
The only safe place to reset the clipboard is between 
SuspendPOEItemScript = 1
and
Send ^{sc02E}

**Comment by Eruyome** [comment]
Found it, there were a few places where this was commented : 

`;SetClipboardContents(ParsedData)`

When you set the clipboard to something here it overwrites/deletes the item data and everything works.

**Comment by Eruyome** [comment]
Feel free to test my changes sometime to make sure everything is fixed and i didn't introduce new issues.

**Comment by 4GForce** [comment]
Soooooo ... It wasn't me =p
or was it ?   I don't remember commenting those lines.
Yeah, SetClipboardContents() looks ok, I'll test more tomorrow.
